sycl/test-e2e/Graph/graph_common.hpp

```diff
@@ -432,3 +432,14 @@ class Barrier {
   std::condition_variable cv;
   std::size_t threadNum;
 };
+
+template <typename T>
```

Comment:
I like the verbosity of this approach. We might want to extend this to check full arrays in the other tests where we only use asserts as well to ease the debugging.

Reply:
This is a good idea, but would be best as a standalone follow-up PR